coderd/templateversions.go

```diff
@@ -60,6 +62,22 @@ func (api *API) templateVersion(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var matchedProvisioners *codersdk.MatchedProvisioners
+	if jobs[0].ProvisionerJob.JobStatus == database.ProvisionerJobStatusPending {
```

Comment:
Do we only ever expect a single job to be returned by `GetProvisionerJobsByIDsWithQueuePosition`?

Reply:
In this instance, we should get either 0 or 1:
```
jobs, err := api.Database.GetProvisionerJobsByIDsWithQueuePosition(ctx, []uuid.UUID{templateVersion.JobID})
```

Reply:
My point here is really that if we're only ever expecting a single job, we should consider changing the semantics of the `GetProvisionerJobsByIDsWithQueuePosition` to be a `:one` not a `:many`.

Reply:
I guess it's _possible_ though that a single template version could have multiple provisioner jobs associated if something went wrong.